sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs

```diff
@@ -0,0 +1,291 @@
+            }
+        }
+
+        [Test]
+        public async Task RecognizeHealthcareEntitiesWithTopParameter()
+        {
+            TextAnalyticsClient client = GetClient();
+
+            HealthcareOptions options = new HealthcareOptions()
+            {
+                Top = 1
+            };
+
+            HealthcareOperation operation = await client.StartHealthcareBatchAsync(batchDocuments, options);
+
+            await operation.WaitForCompletionAsync();
+
+            RecognizeHealthcareEntitiesResultCollection resultCollection = operation.Value;
+
+            Assert.AreEqual(1, resultCollection.Count);
+            Assert.AreEqual(3, resultCollection[0].Entities.Count);
+            Assert.IsNotNull(resultCollection[0].Id);
+            Assert.AreEqual("1", resultCollection[0].Id);
+            Assert.AreEqual("100mg", resultCollection[0].Entities.FirstOrDefault().Text);
+            Assert.AreEqual("Dosage", resultCollection[0].Entities.FirstOrDefault().Category);
+        }
+
+        [Test]
+        public async Task RecognizeHealthcareEntitiesWithSkipParameter()
+        {
+            TextAnalyticsClient client = GetClient();
+
+            HealthcareOptions options = new HealthcareOptions()
+            {
+                Skip = 1
+            };
+
+            HealthcareOperation operation = await client.StartHealthcareBatchAsync(batchDocuments, options);
+
+            await operation.WaitForCompletionAsync();
+
+            RecognizeHealthcareEntitiesResultCollection resultCollection = operation.Value;
+
+            Assert.IsNotNull(resultCollection[0].Warnings);
+            Assert.AreEqual(1, resultCollection.Count);
+            Assert.AreEqual(6, resultCollection[0].Entities.Count);
+            Assert.IsNotNull(resultCollection[0].Id);
+            Assert.AreEqual("2", resultCollection[0].Id);
+            Assert.AreEqual("rapid", resultCollection[0].Entities.FirstOrDefault().Text);
+            Assert.AreEqual("SymptomOrSign", resultCollection[0].Entities.FirstOrDefault().Category);
+        }
+
+        [Test]
+        public async Task RecognizeHealthcareEntitiesBatchWithErrorTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            var documents = new List<string>
+            {
+                "Subject is taking 100mg of ibuprofen twice daily",
+                "Can cause rapid or irregular heartbeat, delirium, panic, psychosis, and heart failure.",
+                "",
+            };
+
+            HealthcareOperation operation = await client.StartHealthcareBatchAsync(documents);
+
+            await operation.WaitForCompletionAsync();
+
+            RecognizeHealthcareEntitiesResultCollection resultCollection = operation.Value;
+
+            Assert.IsNotNull(resultCollection[0].Id);
+            Assert.IsNotNull(resultCollection[0].TextAnalyticsError);
+            Assert.AreEqual("Document text is empty.", resultCollection[0].TextAnalyticsError.Message);
```

Comment:
shouldn't this be `resultCollection[2]` as the wrong document is the 3rd one?

Reply:
The error was coming in the first doc and not the last one. All the tests are passing. Should I send it has List<TextDocument> ?

Reply:
plop... This shouldn't be the case. the result docs need to be in the same order the input was created. that is why we have the createIndexMap function.
Looks like there is some debugging that needs to happen here to understand why the input doesn't match the output